### PR TITLE
Refine RBAC permissions for runner and gateway service accounts

### PR DESCRIPTION
**Pull request by kaivalya97**
Changed the ClusterRoleBindings to use a custom ClusterRole instead of cluster-admin

### CODE REVIEW

File: install-manifests/kubernetes/prod/space-cloud.yaml [hunk 1]
```diff
@@ -55,17 +78,30 @@ metadata:
...
-# This cluster role binding allows anyone in the "manager" group to read secrets in any namespace.
-kind: ClusterRoleBinding
+kind: Role
 metadata:
   name: space-cloud-gateway
+  namespace: space-cloud
+rules:
+  - apiGroups: [""]
+    resources: ["pods"]
+    verbs: ["get", "list"]
```
Comment (YourTechBud) [inline install-manifests/kubernetes/prod/space-cloud.yaml:88]:
```suggestion
    verbs: ["get", "list", "watch"]
```

File: install-manifests/kubernetes/prod/space-cloud.yaml [hunk 2]
```diff
@@ -55,17 +78,30 @@ metadata:
...
+  - apiGroups: [""]
+    resources: ["configmaps", "secrets"]
+    verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
```
Comment (YourTechBud) [inline install-manifests/kubernetes/prod/space-cloud.yaml:94]: The gateway needs cluster access as well apparently.

**Comment by kaivalya97** [comment]
> I think we should make the same changes in the `local` setup as well?

Yes, added

**Comment by YourTechBud** [comment]
Looks good. Let me test this once again tomorrow and then I'll merge it.

File: install-manifests/kubernetes/local/space-cloud.yaml [hunk 3]
```diff
@@ -55,17 +79,28 @@ metadata:
...
 metadata:
   name: space-cloud-gateway
 subjects:
   - kind: ServiceAccount
     name: space-cloud-gateway # Name is case sensitive
-    namespace: space-cloud
 roleRef:
-  kind: ClusterRole
-  name: cluster-admin
+  kind: Role
```
Comment (YourTechBud) [inline install-manifests/kubernetes/local/space-cloud.yaml:102]:
```suggestion
  kind: ClusterRole
```

File: install-manifests/kubernetes/prod/space-cloud.yaml [hunk 4]
```diff
@@ -55,17 +79,28 @@ metadata:
...
 metadata:
   name: space-cloud-gateway
 subjects:
   - kind: ServiceAccount
     name: space-cloud-gateway # Name is case sensitive
-    namespace: space-cloud
 roleRef:
-  kind: ClusterRole
-  name: cluster-admin
+  kind: Role
```
Comment (YourTechBud) [inline install-manifests/kubernetes/prod/space-cloud.yaml:102]:
```suggestion
  kind: ClusterRole
```

**Pull request merged**